src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -313,7 +173,6 @@ public void initializeProfileTransfers() {
         rr.from = new GenericLocation(lat, lon);
         // FIXME requires destination to be set, not necessary for analyst
         rr.to = new GenericLocation(lat, lon);
-        rr.batch = true;
```

Comment:
By removing the `batch` parameter this method uses the point-to-point (not point-to-many) search with the wrong Heuristics function. I made a _quick-fix_ in Entur´s fork in this commit: 567971cb37b54329e90a8bdb9688a4cb593275b1

For OTP 2.0 I plan to do a proper fix by using the StreetSpliter and not the Sample functionality to make it work.

Reply:
I just read through your quick-fix and agree we need to add back the "batch" parameter although I'd like to rename it something like "oneToMany" that better reflects its current usage, and to distinguish it from the existing parameter. The quick fix understandably re-introduces all the code used by the existing `batch` parameter including Sample and SampleFactory and some tile manipulation methods. These are very specific to Analysis, and intended to make the point linking method identical for origins and destinations. With analysis removed, there is no linking of destinations so we should be able to remove the Sample concept and use the same normal origin point linking as in a one-to-one search.

I can re-introduce a simpler version of this quick-fix code before we finally merge this PR. I would also like to add the change that I mentioned in another comment - removing the equals and hashcode methods on RoutingRequest.